openff/arsenic/wrangle.py

```diff
@@ -59,12 +61,109 @@ class ExperimentalResult(object):
     def __init__(self, ligand, expt_DG, expt_dDG):
         self.ligand = ligand
         self.DG = float(expt_DG)
-        self.dDG = float(expt_dDG.strip("\n"))
+        try:
+            error = expt_dDG.strip("\n")
+        except AttributeError:
+            # Data is already numeric
+            error = expt_dDG
+        self.dDG = float(error)
+
+
+class FEResults(object):
+    """
+    Wrapper class to store results from Free Energy calculations.
+
+    Useful for programatically creating a container to pass to FEMap class.
+    """
+
+    def __init__(self):
+        """Create results object"""
+        # Initializing attributes
+        self.experimental_results = dict()
+        self.calculated_results = list()
+        self.fe_results = {
+            "Experimental": self.experimental_results,
+            "Calculated": self.calculated_results,
+        }
+
+    def add_experimental_obj(self, experimental_result):
+        """Adds experimental result from ExperimentalResult object."""
+        self.experimental_results[experimental_result.ligand] = experimental_result
+
+    def add_calculated_obj(self, calculated_result):
+        """Adds calculated result from CalculatedResult object."""
+        self.calculated_results.append(calculated_result)
+
+    def add_experimental_result(self, ligand, expt_value, expt_error):
+        """
+        Add experimental result from plain explicit values.
+
+        Examples
+        --------
+        >>> fe_results = FEResults()
+        >>> fe_results.add_experimental_result("CAT-13a", -8.93, 0.10)
+        """
+        self.experimental_results[ligand] = ExperimentalResult(ligand, expt_value, expt_error)
+
+    def add_calculated_result(self, ligand_a, ligand_b, calc_value, mbar_error, other_error):
+        """
+        Add calculated relative result from plain explicit values.
+
+        Examples
+        --------
+        >>> fe_results = FEResults()
+        >>> fe_results.add_calculated_result("CAT-13a", "CAT-17g", 0.36, 0.11, 0.0)
+        """
+        self.calculated_results.append(
+            RelativeResult(ligand_a, ligand_b, calc_value, mbar_error, other_error)
+        )
 
 
 class FEMap(object):
-    def __init__(self, csv):
-        self.results = read_csv(csv)
+    def __init__(self, input_data):
+        """
+        Construct Free Energy map of simulations from input data.
+
+        Parameters
+        ----------
+            input_data: csv file path or path-like object or dict-like object
+                File path to csv file or python dict-like object.
+
+        Examples
+        --------
+        To read from a csv file specifically formatted for this, you can use:
```

Comment:
Maybe mention here (or link to it if it is in the docs somewhere else) what format the CSV file needs to be in.

Reply:
As far as I can see this format isn't documented. We just have the `data/examples.csv` file that I'm referencing in the example code. I think we could link to that here, I agree. Thanks!

Reply:
We shouldn't link to the file---we should figure out what it is expecting and explicitly specify it here.